src/probnum/randprocs/covfuncs/_covariance_linear_operator.py

```diff
@@ -81,5 +84,9 @@ def _matmul(self, x: np.ndarray) -> np.ndarray:
 
     def _transpose(self) -> linops.LinearOperator:
         return CovarianceLinearOperator(
-            self._x0, self._x1, self._evaluate_dense_matrix, self.keops_lazy_tensor
+            self._x1,
+            self._x0,
+            (self.shape[1], self.shape[0]),
+            self._evaluate_dense_matrix,
+            self._keops_lazy_tensor_fn,
         )
```

Comment:
If I'm not mistaken, then this implementation is only correct if `self` is a actually a covariance function. If `self` is a cross-covariance function, e.g. $\mathcal{L} k$ for some linear operator $\mathcal{L}$, then it might not even be well-defined to swap the inputs like this.

Reply:
Hmm, you're right. Unfortunately, as far as I know, KeOps doesn't allow you to simply get the transpose of a `LazyTensor`. I guess the correct way to do it would then be to add something like a `trans` flag which, if set to `True`, calls `(x.T @ self.keops_lazy_tensor).T` instead of `self.keops_lazy_tensor @ x` in `_matmul`? What do you think?

Reply:
Regarding the transpose of a 'keops.LazyTensor': Does 'mylazytensor.t()' not do what you want? See also: http://www.kernel-operations.io/keops/python/api/common/GenericLazyTensor.html#pykeops.common.lazy_tensor.GenericLazyTensor.t

Reply:
Well, this is awkward. You're absolutely right. I don't know why I didn't find that...
Thanks! Then I'll use this.